datastore/datastore.go

```diff
@@ -125,6 +136,19 @@ func (q *mgoQuery) Select(selector interface{}) Query {
 	return &mgoQuery{q.Query.Select(selector)}
 }
 
+func (q *mgoQuery) Count() (int, error) {
+	return q.Query.Count()
+}
+
+// mgoPipe wraps an mgo.Pipe and implements Pipe
+type mgoPipe struct {
+	mgoP *mgo.Pipe
```

Comment:
Any reason to create a field with the `mgo.Pipe` type here? If you embed the `mgo.Pipe` type (as we have done in `mgoQuery` and `mgoCollection`), you don't need to re-implement the methods unless we want to change the return type (which we don't for `All`).

Reply:
I got confused with an unrelated error and that's why I added this.